xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/Logic.js

```diff
@@ -479,6 +514,91 @@ define([
 
 
 
+    /**
+     * ---------------------------------------------------------------
+     * DISPLAY
+     */
+
+
+    /**
+     * Returns whether a certain property is displayable
+     * @param {String} propertyId
+     * @returns {Boolean}
+     */
+    isPropertyDisplayable: function (propertyId) {
+      var propertyDescriptor = this.getPropertyDescriptor(propertyId);
+      var propertyTypeDescriptor = this.getPropertyTypeDescriptor(propertyId);
+      return (propertyDescriptor.hidden !== undefined && !propertyDescriptor.hidden) ||
+        (propertyDescriptor.hidden === undefined && !propertyTypeDescriptor.hidden);
+    },
+
+
+    /**
+     * Returns the property descriptors of displayable properties
+     * @returns {Array}
+     */
+    getDisplayablePropertyDescriptors: function () {
+      var self = this;
+      return this.data.meta.propertyDescriptors
+      .filter(function (propertyDescriptor) {
+        return self.isPropertyDisplayable(propertyDescriptor.id);
+      })
+      .sort(function (descriptorA, descriptorB) {
+        return self.propertyOrder.indexOf(descriptorA.id) - self.propertyOrder.indexOf(descriptorB.id);
+      });
+    },
+
+
+    /**
+     * Returns whether a certain property is visible
+     * @param {String} propertyId
+     * @returns {Boolean}
+     */
+    isPropertyVisible: function (propertyId) {
+      return this.isPropertyDisplayable(propertyId) &&
+        !this.uniqueArrayHas(this.hiddenProperties, propertyId);
+    },
+
+
+    /**
+     * Set whether the given property should be visible
+     * @param {String} propertyId
+     * @param {Boolean} bool
+     */
+    setPropertyVisibility: function (propertyId, bool) {
+      if (!this.isPropertyDisplayable(propertyId)) { return; }
+      if (bool) {
+        // set visible
+        this.uniqueArrayRemove(this.hiddenProperties, propertyId);
+      } else {
+        // set hidden
+        this.uniqueArrayAdd(this.hiddenProperties, propertyId);
```

Comment:
If we need to watch the visibility from the property descriptor, then the property has to be there from the beginning so that it can be watched by vue. So instead having a `hidden: false` on every properties at the beginning, I changed to `visible: true`, which is more intuitive (and avoid the double negation "!hidden" to know if it is visible)

Reply:
OK, I'll have to update the code on the server side to generate ``visible:true`` instead of ``hidden:false``.